pkg/deviceclaimingserver/grpc_end_devices.go

```diff
@@ -16,16 +16,23 @@ package deviceclaimingserver
 
 import (
 	"context"
+	"net"
 
 	pbtypes "github.com/gogo/protobuf/types"
+	"go.thethings.network/lorawan-stack/v3/pkg/auth/rights"
 	"go.thethings.network/lorawan-stack/v3/pkg/errors"
 	"go.thethings.network/lorawan-stack/v3/pkg/log"
 	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
 	"go.thethings.network/lorawan-stack/v3/pkg/types"
 	"go.thethings.network/lorawan-stack/v3/pkg/web"
 )
 
-var errMethodUnavailable = errors.DefineUnimplemented("method_unavailable", "method available")
+var (
+	errParseQRCode       = errors.Define("parse_qr_code", "parse QR code failed")
+	errQRCodeData        = errors.DefineInvalidArgument("qr_code_data", "invalid QR code data")
+	errNoJoinEUI         = errors.DefineInvalidArgument("no_join_eui", "failed to extract JoinEUI from request")
+	errMethodUnavailable = errors.DefineUnimplemented("method_unavailable", "method available")
```

Comment:
shouldn't it be `method unavailable` in the error message?

If so, the same applies to the file `pkg/deviceclaimingserver/enddevices/util_test.go`. It's only used in the mockEDCS but probably worth a change.